src/Collection/CollectionTrait.php

```diff
@@ -399,6 +399,105 @@ public function last()
         return $result;
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    public function takeLast($howMany)
+    {
+        if ($howMany < 1) {
+            throw new \InvalidArgumentException("The takeLast method requires a number greater than 0.");
+        }
+
+        $iterator = $this->optimizeUnwrap();
+        if (is_array($iterator)) {
+            return new Collection(array_slice($iterator, $howMany * -1));
+        }
+
+        if ($iterator instanceof Countable) {
+            $count = count($iterator);
+
+            if ($count === 0) {
+                return new Collection([]);
+            }
+
+            $iterator = new LimitIterator($iterator, max(0, $count - $howMany), $howMany);
+
+            return new Collection($iterator);
+        }
+
+        $generator = function ($iterator, $howMany) {
```

Comment:
Could we move that to a function or even small class which has your great explanation as doc block?

Reply:
Sure, feel free to send a PR if  you want to work on that